src/capsule/react-native/ReactNativeChallengeStorage.ts

```diff
@@ -0,0 +1,63 @@
+import DeviceCrypto, {
+  AccessLevel,
+  BiometryParams,
+  KeyCreationParams,
+} from 'react-native-device-crypto'
+// @ts-ignore
+import EllipticSignature from 'elliptic/lib/elliptic/ec/signature'
+import { ec } from 'elliptic'
+import { ChallengeStorage, Signature } from '../ChallengeStorage'
+const PEM_HEADER = '-----BEGIN PUBLIC KEY-----'
+const PEM_FOOTER = '-----END PUBLIC KEY-----'
+
+export class ReactNativeChallengeStorage extends ChallengeStorage {
+  protected signOptions(): BiometryParams {
+    return {
+      biometryTitle: 'Authenticate',
+      biometrySubTitle: 'Signing',
+      biometryDescription: 'Authenticate yourself to sign the text',
+    }
+  }
+
+  protected asymmetricKeyOptions(): KeyCreationParams {
+    return {
+      accessLevel: AccessLevel.ALWAYS,
+      invalidateOnNewBiometry: false,
+    }
+  }
+
+  private storageIdentifier() {
+    return 'challenge-' + this.userId
+  }
+
+  async getPublicKey(): Promise<string> {
+    const pemPublicKey = await DeviceCrypto.getOrCreateAsymmetricKey(
+      this.storageIdentifier(),
+      this.asymmetricKeyOptions()
+    )
+
+    const base64PublicKey = pemPublicKey.replace(PEM_FOOTER, '').replace(PEM_HEADER, '').trim()
+    const bufferPublicKey = Buffer.from(base64PublicKey, 'base64')
+    const publicKeyHexAsnPreamble = bufferPublicKey.toString('hex')
+    const publicKeyHex = publicKeyHexAsnPreamble.slice(52)
+    return publicKeyHex
+  }
```

Comment:
From my understanding, `react-native-device-crypto` is itself a bridge for accessing Shared Preferences or Keychain. Is this not going to create its own user prompt, if in `ReactNativePrivateKeyStyorage` we also access the keychain?

Reply:
it will! it is possible, so if you want to reduce the number of prompts you may require lower security settings